File: app/src/main/java/org/wikipedia/suggestededits/provider/SuggestedEditItem.java
package org.wikipedia.suggestededits.provider;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import org.wikipedia.dataclient.restbase.page.RbPageSummary;
import org.wikipedia.gallery.GalleryItem;
import org.wikipedia.wikidata.Entities;

import java.util.Collections;
import java.util.Map;

public class SuggestedEditItem extends RbPageSummary {
    @Nullable @SerializedName("structured") private GalleryItem.StructuredData structuredData;
    @Nullable @SerializedName("wikibase_item") private Entities.Entity entity;

    @NonNull public Map<String, String> getCaptions() {
        return (structuredData != null && structuredData.getCaptions() != null) ? structuredData.getCaptions() : Collections.emptyMap();
    }

    @Nullable public Entities.Entity getEntity() {
        return entity;
    }
}

File: app/src/main/java/org/wikipedia/suggestededits/provider/SuggestedEditItem.kt
package org.wikipedia.suggestededits.provider

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import org.wikipedia.dataclient.wikidata.Entities
import org.wikipedia.gallery.GalleryItem

@Serializable
class SuggestedEditItem {
    private val pageid: Int = 0
    private val ns: Int = 0
    private val title: String? = null
    @SerialName("structured")
    private val structuredData: GalleryItem.StructuredData? = null
    @SerialName("wikibase_item")
    val entity: Entities.Entity? = null

    fun title(): String {
        return title.orEmpty()
    }

    val captions get() = structuredData?.captions ?: emptyMap()
}
